### ISSUE TITLE
Implementing label smoothing.

### ISSUE THREAD

**Issue by dontloo** [fchollet/keras#4704]
Hi,

I'm interested in the label smoothing in [this paper](https://www.robots.ox.ac.uk/~vgg/rg/papers/reinception.pdf). I'm not sure whether we should implement it as a loss function or a pre-processing method for the labels.

At present I've added label smoothing as a choice of pre-processing in the `to_categorical` method in `utils/np_utils.py`, which seems to be the easiest way.

Best regards.
